Runtime/Scripts/RGBotServerListener.cs

```diff
@@ -272,6 +279,13 @@ public void EndClientConnection(long clientId)
             botStateListeners.TryRemove(clientId, out _);
             botStates.TryRemove(clientId, out _);
             mainThreadTaskQueue.TryRemove(clientId, out _);
+            
+            // If there are no more bots, we can clear all of the data used so far
+            if (clientConnectionMap.IsEmpty)
+            {
+                // TODO: This seems to call early due to the recording being a task?
+                //_dataCollection.Cleanup();
```

Comment:
Any ideas here? Because the upload functionality is async, this gets called before RecordSession is finished. Maybe I'll try to add a wait for tasks call in the RecordSession function somehow.

Reply:
If you make `RecordSession` an `async Task` method, you can `await` that `Task`. That will ensure the rest of the method doesn't execute until all the async operations `await`ed in `RecordSession` complete. It'll mean the execution of this method will spread over several frames, but if you want to wait for the screenshots to upload, that's what you need to do.

Reply:
confirmed that Aaron had made these updates ... resolving this as ashley is out for a few days